src/renderer.ts

```diff
@@ -107,67 +94,74 @@ export class Renderer {
    *   - a div element, which will contain the SVG output
    * @param backend Renderer.Backends.CANVAS or Renderer.Backends.SVG
    */
-  constructor(canvasId: string | HTMLCanvasElement | HTMLDivElement, backend: number) {
-    if (!canvasId) {
-      throw new RuntimeError('BadArgument', 'Invalid id for renderer.');
-    } else if (typeof canvasId === 'string') {
-      this.elementId = canvasId;
-      this.element = document.getElementById(canvasId) as HTMLCanvasElement | HTMLDivElement;
-    } else if ('getContext' in canvasId /* HTMLCanvasElement */) {
-      this.element = canvasId as HTMLCanvasElement;
+  constructor(context: RenderContext);
+  constructor(canvas: string | HTMLCanvasElement | HTMLDivElement, backend: number);
+  constructor(arg0: string | HTMLCanvasElement | HTMLDivElement | RenderContext, arg1?: number) {
+    if (isRenderContext(arg0)) {
+      // The user has provided what looks like a RenderContext, let's just use it.
+      // TODO(tommadams): RenderContext is an interface, can we introduce a context base class
```

Comment:
I wouldn't mind having an abstract base RenderContext class that people are required to extend. 

Alternatively, we go the isCategory route (like our type guard file) where if you have a static property CATEGORY that returns 'RenderContext', we will trust that you know what you're doing. 

@0xfe any input on this? I'd prefer an abstract class that both SVGContext and CanvasContext extend from.